.github/workflows/create-release.yml

```diff
@@ -58,28 +49,66 @@ jobs:
         conda deactivate
         echo "# Platform-specific torch requirements (See SCT Issue #2745)" >> requirements-freeze.txt
         grep "torch" requirements.txt >> requirements-freeze.txt
```

Comment:
Maybe out of scope, but I'd be curious to see how `grep "torch"` interacts with the recent changes in #3790 (which remove the lines that used to begin with "torch", and adds an "--extra-index-url" line which happens to contain the string "torch" in its url).

Reply:
Thank you for keeping an eye on this!

Thankfully, it works as we'd want it to:

```console
joshua@tadpole:~/repos/spinalcordtoolbox$ grep "torch" requirements.txt
--extra-index-url https://download.pytorch.org/whl/cpu
```

But, that almost feels by luck, rather than by design? :P 

Grepping within `requirements.txt` files feels so odd...

Reply:
One other nice bonus of using the `upload-artifact` and `download-artifact` actions is that it actually creates file download [within the workflow itself](https://github.com/spinalcordtoolbox/spinalcordtoolbox/actions/runs/2651333502):

![image](https://user-images.githubusercontent.com/16181459/178355881-0e62346b-d538-4b2a-90e3-13e8933753ec.png)

Downloading that file shows us the same result:

![image](https://user-images.githubusercontent.com/16181459/178356068-924ad52f-012d-47e2-8bef-917de7a34a4b.png)

Reply:
Ah, but the echo'd comment *is* out of date -- we're no longer concerned with platform-specific torch requirements.

Reply:
I'm glad it's intentional! I did get the feeling that luck might be involved, hence the question.

I think grepping through config files is a time-honored tradition, although there are ways to make it more robust. Purely as an example, since [pip supports end-of-line comments](https://pip.pypa.io/en/latest/reference/requirements-file-format/#comments), we could instead grep for a token like `append_to_freeze` that we would put in an end-of-line comment on the lines we want.

Reply:
> Purely as an example, since [pip supports end-of-line comments](https://pip.pypa.io/en/latest/reference/requirements-file-format/#comments), we could instead grep for a token like append_to_freeze that we would put in an end-of-line comment on the lines we want.

Ah! I absolutely love this idea! There are all sorts of benefits here:

* `create-release.yml` only needs a single grep pattern (i.e. it doesn't need to be continuously updated)
* So, any future changes would only occur within the `requirements.txt` file, which is where we're already making our changes, giving us a single location to focus on (preventing us from accidentally missing the workflow file when making changes to the requirements file)
* The pattern can be chosen to be unique enough to avoid accidental/lucky pattern-matching